org.mixedrealitytoolkit.input/InteractionModes/NearInteractionModeDetector.cs

```diff
@@ -25,49 +28,68 @@ public class NearInteractionModeDetector : ProximityDetector
 
         /// <inheritdoc />
         public override bool IsModeDetected()
         {
             bool result = base.IsModeDetected() || IsNearInteractorSelecting();
             if (result)
             {
-                for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+                UpdateProximityExited();
+                UpdateProximityEntered();
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// Call OnProximityExited for all colliders that were previously detected but are no longer detected.  This
+        /// effectively triggers OnProximityExited for all colliders that are no longer detected.
+        /// </summary>
+        private void UpdateProximityExited()
+        {
+            for (int i = 0; i < previouslyDetectedColliders.Count; i++)
+            {
+                Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
+                if (!DetectedColliders.Contains(previouslyDetectedCollider) &&
+                    InteractionManager.TryGetInteractableForCollider(previouslyDetectedCollider, out IXRInteractable xrInteractable) &&
+                    xrInteractable is IXRProximityInteractable xrProximityInteractable)
                 {
-                    Collider previouslyDetectedCollider = previouslyDetectedColliders[i];
-                    if (!DetectedColliders.Contains(previouslyDetectedCollider) && previouslyDetectedCollider != null)
+                    foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
                     {
-                        IXRProximityInteractable nearInteractionMode = previouslyDetectedCollider.GetComponent<IXRProximityInteractable>();
-                        if (nearInteractionMode != null)
-                        {
-                            foreach (XRBaseInteractor xrBaseInteractor in nearInteractors)
-                            {
-                                previouslyDetectedCollider.GetComponent<IXRProximityInteractable>().OnProximityExited(new ProximityExitedEventArgs(this, previouslyDetectedCollider, xrBaseInteractor));
-                            }
-                        }
-                        previouslyDetectedColliders.Remove(previouslyDetectedCollider);
+                        xrProximityInteractable.OnProximityExited(new ProximityExitedEventArgs(previouslyDetectedCollider, xrBaseInteractor));
                     }
+                    previouslyDetectedColliders.Remove(previouslyDetectedCollider);
                 }
-                foreach (Collider collider in DetectedColliders)
+            }
+        }
+
+        /// <summary>
+        /// Calll OnProximityEntered for all colliders that are currently detected but were not detected previously.  This
+        /// effectively triggers OnProximityEntered for all colliders that are newly detected.
+        /// </summary>
+        private void UpdateProximityEntered()
+        {
+
```

Comment:
nit.  remove extra new line

Reply:
Whoops, my bad.  Removed in latest commit, resolving.